### PR TITLE
Splitting out dns from http

### PR DESCRIPTION
**Pull request by AndyG-0**
This allows for better compatibility with Traefik on k3s. Splitting out the http traffic allows you to use clusterIP for the http service which can be used by the ingress. Right now as things stand you have to select one or the other for all TCP traffic. So if you have Traefik running the 80/443 is blocked for the loadbalancer since it's already used by Traefik. Splitting this out allows the option to do either and makes it more compatible. I have tested this out in my local k3s cluster and it works with Traefik. Because I'm using Traefik I can't test with all load balancers but I believe it should work fine.

### CODE REVIEW

**Comment by MoJo2600** [comment]
Looks good, thank you! but i will test it with metallb, but i don't see why this should not work.

**Comment by chrismeller** [comment]
I can also confirm that this is working well for me on k3s with Traefik, though I would suggest changing the naming to indicate that TCP is actually TCP-HTTP

**Comment by MoJo2600** [comment]
@AndyG-0 I had to do some minor modifications to get it working with metallb and updated the documentation. I pushed my changes to the branch [AndyG-0-master](https://github.com/MoJo2600/pihole-kubernetes/tree/AndyG-0-master). Maybe you could add these canges to your pull request? 

One question: When we split up the services, is this a breaking change to the existing installations? At least for meatllb you have to specify the IP for the new service. If it is the same for Traefik,  we should maybe bump the version to 1.8 and add a disclaimer about this to the Readme. What do you think?

**Comment by AndyG-0** [comment]
@MoJo2600 I haven't forgotten about this. I have some time off coming up that I'm hoping to get back to this to be able to update.

**Comment by AndyG-0** [comment]
Merged changes and works fine in k3s with one tweak to ingress.yaml to reflect service name changes. I also had to revert the chart.yaml maintainer name to MoJo2600 to pass linting. I can change it back if the linter can be changed. I would not think this is a breaking change as it still works the same as it did before it just works better now with traefik and an ingress.

**Comment by MoJo2600** [comment]
Ok, thank you for doing the changes. I will try it on my cluster once more and then merge it. Thank you very much. Good work!

**Comment by dza89** [comment]
I think it makes more sense to merge the DNS values in the valuefile. Since both services are part of a single configuration.
@AndyG-0  @MoJo2600 

Let me know.

https://github.com/AndyG-0/pihole-kubernetes/pull/1/files

**Comment by MoJo2600** [comment]
I do like those changes, and +1 for `apiVersion: networking.k8s.io/v1` does this still fit the traefix use case? I assume it does @AndyG-0?

**Comment by dza89** [comment]
According to lower camel case. Abbreviations of 3 or more chars should be lowercase. So I'll make one more commit later today.

**Comment by AndyG-0** [comment]
The changes are good for the most part but I think the ingress change should be discussed. I sadly have fell behind in keeping my k3s up to date. The ingress change was introduced in 1.19 but I'm on 1.18 so the new ingress definition fails for me. Shame on me for being behind but this does mean that the ingress will not work on previous versions of Kubernetes prior to 1.19. It's probably better to support the go forward version since the old version is deprecated. I just want to make sure everyone is ok with the implications. Thoughts? Side note I did test by reverting the api version in the ingress and changing the backend portion and it worked fine.

**Comment by dza89** [comment]
@AndyG-0 
You are right, i've changed it to networking v1beta1, which is supported since 1.14.
Can you try it now?
I don't think any helm chart should support below 1.14.

**Comment by AndyG-0** [comment]
@MoJo2600 What are your thoughts on this? One approach is to keep it with the old deprecated api version so that it works on against the older kube api. The other would be to switch it to the new to get off the deprecated version and then put something in the readme about how to get it to work on the older version if necessary. Thoughts?

**Comment by MoJo2600** [comment]
I think i would add the new API version. If the old version is deprecated we can remove it. I bumped the major version and we should add a disclaimer with the minimum version the chart runs on. Would you add this disclaimer?

Good for me, now i have an incentive to upgrade my cluster as well... :)

**Comment by AndyG-0** [comment]
Merged @dza89 changes. Thank You. Changed it back to the newer API and updated the readme. Tested on my newly upgraded k3s 1.19.4 cluster and working fine. Feel free to change the wording in the readme if it's not to your liking.

**Comment by MoJo2600** [comment]
I moved the changes in the README to the gotmpl file and regenerated the README.md. So the values section is always up to date. Thank you @AndyG-0 and @dza89 for your contribution!

**Comment by Starefossen** [comment]
According to SemVer this should have been marked as a breaking change.

File: charts/pihole/Chart.yaml [hunk 1]
```diff
@@ -3,8 +3,9 @@ appVersion: "5.2"
 description: Installs pihole in kubernetes
 home: https://github.com/MoJo2600/pihole-kubernetes/tree/master/charts/pihole
 name: pihole
-version: 1.7.21
+version: 1.8.22
```
Comment (Starefossen) [inline charts/pihole/Chart.yaml:6]: This should have been 2.x

Comment (MoJo2600) [reply charts/pihole/Chart.yaml:6]: Yes, you are right, now as I think about it. Changing it now would not mehr sense, I suppose. Have to keep this in mind for the next changes.

**Comment by jjaybrown** [comment]
Can you share your traefik config/yaml trying to get this to work with `dns.mydomain`

**Pull request merged**